packages/framework/src/handler.ts

```diff
@@ -176,8 +176,21 @@ export class NovuRequestHandler<Input extends any[] = any[], Output = any> {
     action: string
   ): Record<PostActionEnum, () => Promise<IActionResponse>> {
     return {
-      [PostActionEnum.TRIGGER]: this.triggerAction({ workflowId, ...body }),
+      [PostActionEnum.TRIGGER]: async () => {
+        const errors = this.client.getErrors();
+
+        if (errors.length !== 0) {
+          return this.createResponse(HttpStatusEnum.UNPROCESSABLE_ENTITY, errors);
+        }
+
+        return this.triggerAction({ workflowId, ...body })();
+      },
       [PostActionEnum.EXECUTE]: async () => {
+        const errors = this.client.getErrors();
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** ‏ what is the usecase for execute? I'm not sure how it gets used in the studio and the bridge api client doesn't seem to use the action.

Reply:
cc: @rifont ☝️